### PR TITLE
feat(SW): Implement low mem variant

### PR DESCRIPTION
**Pull request by DanielVoogsgerd**
This implementation uses a 2D circular buffer as the underlying datastructure. We can theoretically determine that the maximum length of a traceback can at most be:

```
wrapping_height =  |query| * (1 + (MATCH_SCORE / GAP_PENALTY))
```

by using a circular buffer we the height can be wrapping height. Considering the query is often *a lot* smaller than the target this will drastically reduce the size of the matrix.

A quick calculation arises at the following size of the matrix
```
matrix_size = width * wrapping_height * size of integer
```

In our case:
```
matrix_size = 331 * ( 330 * (1 + (2 / 1))) * 16 bit = 655kB
```

There is a downside to this algorithm however. As soon as we find a new best score we have to do the traceback right away because otherwise we could overwrite the data on the next line.
Because of this there is a branch instruction inside the hot loop which will come with some performance cost. There are micro-optimizations possible, but maybe not right now.

Also all index calculations now use a mod instruction which are not necessary in the original algorithm.

In conclusion, use this algorithm when memory might be constraint. Maybe worker nodes can specify a upper limit of resources that can be used and if a job requires more than that using the original method we can switch to this implementation.

### CODE REVIEW

**Comment by DanielVoogsgerd** [comment]
Implemented, and merged somehow
